### PR TITLE
chore(application-system): Application system/introduce statuses to applications

### PR DESCRIPTION
**Pull request by Thorrik58**
# Introduce statuses to applications and change FormModes to same statuses.

Attach a link to issue if relevant

## What

Add application statuses to applications template and adjust FormModes on all forms.

## Why

To form a of standard on all application statuses and form modes.

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Additional info

You'll need to run the migration in order to make new applications if you want to try it out.
Run `yarn nx run application-system-api:migrate`

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by obmagnusson — APPROVED** [review]
LGTM

File: libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts [hunk 1]
```diff
@@ -561,6 +571,7 @@ const ParentalLeaveTemplate: ApplicationTemplate<
         entry: 'assignToVMST',
         meta: {
           name: States.APPROVED,
+          status: 'approved',
```
Comment (qdd1) [inline libs/application/templates/parental-leave/src/lib/ParentalLeaveTemplate.ts:574]: Status is still 'inprogress' here. Even VMST has approved application. Applicant could still add information/make changes and send it in again

**Review by HelgiHH — APPROVED** [review]
fsi files lgtm

**Review by qdd1 — APPROVED** [review]
except the comment, lgtm

File: libs/application/templates/financial-aid/src/forms/Spouse.ts [hunk 2]
```diff
@@ -14,7 +14,7 @@ import { Routes } from '../lib/constants'
 export const Spouse: Form = buildForm({
   id: 'FinancialAidApplication',
   title: m.application.name,
-  mode: FormModes.APPLYING,
+  mode: FormModes.DRAFT,
```
Comment (davidpxn) [inline libs/application/templates/financial-aid/src/forms/Spouse.ts:17]: This should be IN_PROGRESS to be consistent with the status in the template

**Review by davidpxn — CHANGES_REQUESTED** [review]
Just one comment, otherwise the Financial aid files look good 👌

**Review by eirikurn — APPROVED** [review]
Approving core files 🙌

**Review by kksteini — APPROVED** [review]

**Comment by Thorrik58** [comment]
@johannaagma and @davidpxn can you verify that your comments have been addressed?

**Review by davidpxn — APPROVED** [review]

**Pull request merged**